src/operator/custom/custom-inl.h

```diff
@@ -172,10 +186,16 @@ class CustomOperator {
       cv_.wait(lock, [&] {return !q_.empty() || destructing_;});
       while (!q_.empty()) {
         --num_free_threads;
-        auto fn = q_.front();
+        auto task = q_.front();
         q_.pop();
         lock.unlock();
-        fn();
+        try {
+          task.fn();
+        } catch (dmlc::Error& e) {
```

Comment:
What about other types of exceptions? Could we add a comment in the code clarifying this?

Reply:
What other types? I think the only valid exception is dmlc::Error, other exception means wrong code.